x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/extractor/ExtractedFields.java

```diff
@@ -235,39 +235,62 @@ private static boolean isNestedOrObject(Map<String, FieldCapabilities> fieldCaps
     /**
      * Makes boolean fields behave as a field of different type.
      */
-    private static final class BooleanMapper<T> extends DocValueField {
+    private static final class BooleanMapper<T> extends AbstractField {
 
         private static final Set<String> TYPES = Collections.singleton(BooleanFieldMapper.CONTENT_TYPE);
 
+        private final ExtractedField field;
         private final T trueValue;
         private final T falseValue;
 
         BooleanMapper(ExtractedField field, T trueValue, T falseValue) {
             super(field.getName(), TYPES);
-            if (field.getMethod() != Method.DOC_VALUE || field.getTypes().contains(BooleanFieldMapper.CONTENT_TYPE) == false) {
+            this.field = field;
+            if (field.getTypes().contains(BooleanFieldMapper.CONTENT_TYPE) == false) {
                 throw new IllegalArgumentException("cannot apply boolean mapping to field [" + field.getName() + "]");
             }
             this.trueValue = trueValue;
             this.falseValue = falseValue;
         }
 
+        @Override
+        public Method getMethod() {
+            return field.getMethod();
+        }
+
         @Override
         public Object[] value(SearchHit hit) {
-            DocumentField keyValue = hit.field(getName());
-            if (keyValue != null) {
-                return keyValue.getValues().stream().map(v -> Boolean.TRUE.equals(v) ? trueValue : falseValue).toArray();
+            Object[] value = field.value(hit);
+            if (value != null) {
+                return Arrays.stream(value).map(v -> Boolean.TRUE.equals(v) ? trueValue : falseValue).toArray();
             }
```

Comment:
I am not 100% sure this works, in `_source` the boolean representation could be `"true"`.

Reply:
Well spotted! I'm now using `Booleans.parseBoolean` which is also what's used by the `BooleanFieldMapper` when the values are strings.

Reply:
Somehow I forgot to push that change. It should be available now.